graylog2-server/src/main/java/org/graylog2/indexer/results/FieldStatsResult.java

```diff
@@ -28,60 +29,67 @@
 public class FieldStatsResult extends IndexQueryResult {
 
     @Nullable
+    private final ValueCount valueCount;
+    @Nullable
     private final ExtendedStats extendedStats;
     @Nullable
     private final Cardinality cardinality;
     private List<ResultMessage> searchHits;
 
 
-    public FieldStatsResult(@Nullable ExtendedStats extendedStats,
+    public FieldStatsResult(@Nullable ValueCount valueCount,
+                            @Nullable ExtendedStats extendedStats,
                             @Nullable Cardinality cardinality,
                             SearchHits hits,
                             String query,
                             BytesReference source, TimeValue took) {
         super(query, source, took);
-
-
+        this.valueCount = valueCount;
         this.extendedStats = extendedStats;
         this.cardinality = cardinality;
         this.searchHits = buildResults(hits);
     }
 
     public long getCount() {
-        return extendedStats != null ? extendedStats.getCount() : 0;
+        if (valueCount != null) {
+            return valueCount.getValue();
+        } else if (extendedStats != null) {
+            return extendedStats.getCount();
+        }
+        return Long.MIN_VALUE;
```

Comment:
Shouldn't we return 0 here instead of `Long.MIN_VALUE` when the count is not available?